src/globus_sdk/utils.py

```diff
@@ -22,6 +22,41 @@
     PayloadWrapperBase = collections.UserDict
 
 
+class MissingType:
+    def __init__(self) -> None:
+        # disable instantiation, but gated to be able to run once
+        # when this module is imported
+        if "MISSING" in globals():
+            raise TypeError("MissingType should not be instantiated")
+
+    def __bool__(self) -> bool:
+        return False
+
+    def __copy__(self) -> MissingType:
+        return self
+
+    def __deepcopy__(self, memo: dict[int, t.Any]) -> MissingType:
+        return self
+
+    # unpickling a MissingType should always return the "MISSING" sentinel
+    def __reduce__(self) -> str:
+        return "MISSING"
+
+    def __repr__(self) -> str:
+        return "<globus_sdk.MISSING>"
+
+
+# a sentinel value for "missing" values which are distinguished from `None` (null)
+# this is the default used to indicate that a parameter was not passed, so that
+# method calls passing `None` can be distinguished from those which did not pass any
+# value
+# users should typically not use this value directly, but it is part of the public SDK
+# interfaces along with its type for annotation purposes
+#
+# *new after v3.29.0* (older methods do not use MISSING)
```

Comment:
```suggestion
# *new after v3.29.0* (older versions do not use MISSING)
```

Reply:
I actually meant to indicate that methods which were introduced prior to MISSING don't use it (as opposed to those introduced after it). But on consideration, I think the whole parenthetical is just distracting, so I'm striking it.